src/frontend/src/lib/derived/exchange.derived.ts

```diff
@@ -62,7 +43,26 @@ export const exchanges: Readable<ExchangesData> = derived(
 									? icpPrice
 									: undefined
 				};
-			}, {})
+			}, {}),
+			...Object.entries($exchangeStore ?? {}).reduce((acc, [key, currentPrice]) => {
+				const token =
```

Comment:
well, we are iterating first on the keys and values of the exchange store record, so we would not know beforehand if it is erc20 or icrc, unless we change the keys of the exchange store to save this info too.

However, I think that would be better if we refactor everything, so that we loop only once in `erc20Tokens` and only once in `icrcTokens` (right now we do it twice for both).

If it is ok, I would do it in another PR, I prefer to separate the two

Reply:
But we are already iterating on `...$icrcTokens.reduce((acc, token)`. I'm actually a bit confused about the proposed change here. I probably need to give it a try to understand better. Will do Monday and report.

Reply:
I think i understood now your point. have a look at my last commit:

```typescript
                            ...$icrcTokens.reduce((acc, token) => {
				const { id, ledgerCanisterId, exchangeCoinId } = token;

				const icrcPrice = $exchangeStore?.[ledgerCanisterId];

				if (nonNullish(icrcPrice)) {
					return {
						...acc,
						[id]: icrcPrice
					};
				}

				const { twinToken } = token as Partial<IcCkToken>;
                                 [rest of the code]
```

Reply:
That's better for performance, thanks